common/persistence/tests/history_store.go

```diff
@@ -108,14 +110,15 @@ func (s *HistoryEventsSuite) SetupTest() {
 	s.Assertions = require.New(s.T())
 	s.ProtoAssertions = protorequire.New(s.T())
 	s.Ctx, s.Cancel = context.WithTimeout(context.Background(), 30*time.Second*debug.TimeoutMultiplier)
+
+	s.ShardID++
```

Comment:
we have to use different shardIDs for each test? I was under the impression that those tests won't impact each other.

Reply:
There's precedence in this monotonically increasing pattern in the mutable state tests (see https://github.com/temporalio/temporal/blob/main/common/persistence/tests/execution_mutable_state.go#L108)